src/read_utils.rs

```diff
@@ -102,6 +102,23 @@ pub fn set_dataset_validity_period(
     Ok(())
 }
 
+#[cfg(feature = "proj")]
+pub fn update_validity_period(
```

Comment:
I'm not sure this is the good place
a "write" operation in s read module

Reply:
Will be refactored soon